### DISCUSSION TITLE
Question about Optimization Levels

### DISCUSSION THREAD

**Discussion by p-bakker** [mozilla/rhino#1123] [Q&A]
Rhino supports optimization level -1 through 9.

Am trying to check old documentation for porting over to our new documentation and trying to find what the different optimization levels do, but I cannot find any difference between optLevel 1 through 9: nowhere in the codebase can I find any code that acts on any of the specific values 1 through 9. The only thing I can find are if statements that check whether the optLevel > 0

Am I missing something somewhere?

**Comment by C-Masuda** [comment]
I've never seen any explained difference other than what's available in the [old doc via wayback machine](https://web.archive.org/web/20210111111751/https://developer.mozilla.org/en-US/docs/Mozilla/Projects/Rhino/Optimization).
Edit: which is exactly what you've got on your placeholder repo.
